code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -420,13 +420,18 @@
 	unique_pet = TRUE
 	var/age = 0
 	var/record_age = 1
-	var/memory_saved = FALSE
 	var/saved_head //path
 
 /mob/living/simple_animal/pet/dog/corgi/Ian/Initialize(mapload)
 	. = ..()
-	//parent call must happen first to ensure IAN
-	//is not in nullspace when child puppies spawn
+	SSpersistent_data.register(src)
+
+/mob/living/simple_animal/pet/dog/corgi/Ian/death()
+	Write_Memory(TRUE)
+	SSpersistent_data.registered_atoms -= src // We already wrote here, dont overwrite!
+	..()
```

Comment:
I'd put this before the write memory etc. It will return FALSE if it did not die and TRUE if it did die.
So also return the value

Reply:
I will look at other implementations before I start changing things round. This was meant to only change how reading and writing was called, not refactor death

Reply:
It's not refactoring death. It's just taking the value that it returns and using that to check if the dog actually died

Reply:
```
/mob/living/simple_animal/pet/dog/corgi/Ian/death()
	. = ..()
	if(.)
		Write_Memory(TRUE)
		SSpersistent_data.registered_atoms -= src // We already wrote here, dont overwrite!

```
???